Inedo.DependencyScan/NpmDependencyScanner.cs

```diff
@@ -22,13 +22,41 @@ public override async Task<IReadOnlyCollection<ScannedProject>> ResolveDependenc
 
         private static IEnumerable<DependencyPackage> ReadDependencies(JsonDocument doc)
         {
-            if (!doc.RootElement.TryGetProperty("dependencies", out var dependencies))
+            if (!doc.RootElement.TryGetProperty("lockfileVersion", out var lockFileVersion))
                 yield break;
 
-            foreach (var d in dependencies.EnumerateObject())
+            var lockFileVersionString = lockFileVersion.ToString();
+
+            JsonElement npmDependencyPackages;
+            if (lockFileVersionString.Equals("3"))
+            {
+                if (!doc.RootElement.TryGetProperty("packages", out npmDependencyPackages))
+                    yield break;
+            }
+            else 
+            {
+                if (!doc.RootElement.TryGetProperty("dependencies", out npmDependencyPackages))
+                    yield break;
+            }
+
+            foreach (var npmDependencyPackage in npmDependencyPackages.EnumerateObject())
             {
-                string name = d.Name;
-                var version = d.Value.GetProperty("version").GetString();
+                string name = string.Empty;
+                if (lockFileVersionString.Equals("3"))
+                {
+                    // skip the self reference package
```

Comment:
Skipping the root package, [per comment on the forum thread](https://forums.inedo.com/topic/3877/pgscan-lockfileversion-3-for-npm-dependencies-not-supported?_=1690565725743).